File: app/src/main/java/com/dev/nicola/allweather/weatherProvider/DarkSky/model/DailyData.java
package com.dev.nicola.allweather.weatherProvider.DarkSky.model;

import com.google.gson.annotations.SerializedName;

@SuppressWarnings({"UnusedDeclaration"})
class DailyData {

    @SerializedName("time")
    private Integer time;
    @SerializedName("summary")
    private String summary;
    @SerializedName("icon")
    private String icon;
    @SerializedName("sunriseTime")
    private Integer sunriseTime;
    @SerializedName("sunsetTime")
    private Integer sunsetTime;
    @SerializedName("moonPhase")
    private Double moonPhase;
    @SerializedName("precipIntensity")
    private Double precipIntensity;
    @SerializedName("precipIntensityMax")
    private Double precipIntensityMax;
    @SerializedName("precipIntensityMaxTime")
    private Integer precipIntensityMaxTime;
    @SerializedName("precipProbability")
    private Double precipProbability;
    @SerializedName("precipType")
    private String precipType;
    @SerializedName("temperatureMin")
    private Double temperatureMin;
    @SerializedName("temperatureMinTime")
    private Integer temperatureMinTime;
    @SerializedName("temperatureMax")
    private Double temperatureMax;
    @SerializedName("temperatureMaxTime")
    private Integer temperatureMaxTime;
    @SerializedName("apparentTemperatureMin")
    private Double apparentTemperatureMin;
    @SerializedName("apparentTemperatureMinTime")
    private Integer apparentTemperatureMinTime;
    @SerializedName("apparentTemperatureMax")
    private Double apparentTemperatureMax;
    @SerializedName("apparentTemperatureMaxTime")
    private Integer apparentTemperatureMaxTime;
    @SerializedName("dewPoint")
    private Double dewPoint;
    @SerializedName("humidity")
    private Double humidity;
    @SerializedName("windSpeed")
    private Double windSpeed;
    @SerializedName("windBearing")
    private Integer windBearing;
    @SerializedName("visibility")
    private Double visibility;
    @SerializedName("cloudCover")
    private Double cloudCover;
    @SerializedName("pressure")
    private Double pressure;
    @SerializedName("ozone")
    private Double ozone;

    public DailyData() {}

    public Integer getTime() {
        return time;
    }

    public String getSummary() {
        return summary;
    }

    public String getIcon() {
        return icon;
    }

    public Integer getSunriseTime() {
        return sunriseTime;
    }

    public Integer getSunsetTime() {
        return sunsetTime;
    }

    public Double getMoonPhase() {
        return moonPhase;
    }

    public Double getPrecipIntensity() {
        return precipIntensity;
    }

    public Double getPrecipIntensityMax() {
        return precipIntensityMax;
    }

    public Integer getPrecipIntensityMaxTime() {
        return precipIntensityMaxTime;
    }

    public Double getPrecipProbability() {
        return precipProbability;
    }

    public String getPrecipType() {
        return precipType;
    }

    public Double getTemperatureMin() {
        return temperatureMin;
    }

    public Integer getTemperatureMinTime() {
        return temperatureMinTime;
    }

    public Double getTemperatureMax() {
        return temperatureMax;
    }

    public Integer getTemperatureMaxTime() {
        return temperatureMaxTime;
    }

    public Double getApparentTemperatureMin() {
        return apparentTemperatureMin;
    }

    public Integer getApparentTemperatureMinTime() {
        return apparentTemperatureMinTime;
    }

    public Double getApparentTemperatureMax() {
        return apparentTemperatureMax;
    }

    public Integer getApparentTemperatureMaxTime() {
        return apparentTemperatureMaxTime;
    }

    public Double getDewPoint() {
        return dewPoint;
    }

    public Double getHumidity() {
        return humidity;
    }

    public Double getWindSpeed() {
        return windSpeed;
    }

    public Integer getWindBearing() {
        return windBearing;
    }

    public Double getVisibility() {
        return visibility;
    }

    public Double getCloudCover() {
        return cloudCover;
    }

    public Double getPressure() {
        return pressure;
    }

    public Double getOzone() {
        return ozone;
    }
}

File: app/src/main/java/com/dev/nicola/allweather/weatherProvider/DarkSky/model/DailyData.kt
package com.dev.nicola.allweather.weatherProvider.DarkSky.model

import com.google.gson.annotations.SerializedName

class DailyData {

    @SerializedName("time")
    val time: Int? = null
    @SerializedName("summary")
    val summary: String? = null
    @SerializedName("icon")
    val icon: String? = null
    @SerializedName("sunriseTime")
    val sunriseTime: Int? = null
    @SerializedName("sunsetTime")
    val sunsetTime: Int? = null
    @SerializedName("moonPhase")
    val moonPhase: Double? = null
    @SerializedName("precipIntensity")
    val precipIntensity: Double? = null
    @SerializedName("precipIntensityMax")
    val precipIntensityMax: Double? = null
    @SerializedName("precipIntensityMaxTime")
    val precipIntensityMaxTime: Int? = null
    @SerializedName("precipProbability")
    val precipProbability: Double? = null
    @SerializedName("precipType")
    val precipType: String? = null
    @SerializedName("temperatureMin")
    val temperatureMin: Double? = null
    @SerializedName("temperatureMinTime")
    val temperatureMinTime: Int? = null
    @SerializedName("temperatureMax")
    val temperatureMax: Double? = null
    @SerializedName("temperatureMaxTime")
    val temperatureMaxTime: Int? = null
    @SerializedName("apparentTemperatureMin")
    val apparentTemperatureMin: Double? = null
    @SerializedName("apparentTemperatureMinTime")
    val apparentTemperatureMinTime: Int? = null
    @SerializedName("apparentTemperatureMax")
    val apparentTemperatureMax: Double? = null
    @SerializedName("apparentTemperatureMaxTime")
    val apparentTemperatureMaxTime: Int? = null
    @SerializedName("dewPoint")
    val dewPoint: Double? = null
    @SerializedName("humidity")
    val humidity: Double? = null
    @SerializedName("windSpeed")
    val windSpeed: Double? = null
    @SerializedName("windBearing")
    val windBearing: Int? = null
    @SerializedName("visibility")
    val visibility: Double? = null
    @SerializedName("cloudCover")
    val cloudCover: Double? = null
    @SerializedName("pressure")
    val pressure: Double? = null
    @SerializedName("ozone")
    val ozone: Double? = null
}
